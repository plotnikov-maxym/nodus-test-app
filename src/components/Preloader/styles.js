export const styles = () => ({
  root: {
    flexGrow: 1,
    position: "absolute",
    top: 0,
    width: "100%",
  },
});
