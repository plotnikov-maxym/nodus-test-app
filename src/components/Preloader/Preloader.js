import React from "react";
import LinearProgress from "@material-ui/core/LinearProgress";
import {bool, object, string} from "prop-types";
import {styles} from "./styles";
import {withStyles} from "@material-ui/core/styles";

export const Preloader = ({classes, color, loading}) => {
  return (
    loading && (
      <div className={classes.root}>
        <LinearProgress color={color} />
      </div>
    )
  );
};

Preloader.propTypes = {
  classes: object.isRequired,
  loading: bool.isRequired,
  color: string,
};

export default withStyles(styles)(Preloader);
