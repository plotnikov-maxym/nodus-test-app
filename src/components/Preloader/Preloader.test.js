import React from "react";
import renderer from "react-test-renderer";
import {extend} from "underscore";
import {Preloader} from "./Preloader";
import {createShallow} from "@material-ui/core/test-utils";
import LinearProgress from "@material-ui/core/LinearProgress";

describe("<Preloader />", () => {
  let shallow;

  beforeAll(() => {
    shallow = createShallow();
  });

  describe("Preloader component", () => {
    it("matches snapshot", async () => {
      const wrapper = await renderComponent();
      expect(wrapper).toMatchSnapshot();
    });

    it("renders the correct color", async () => {
      const color = "secondary";
      const wrapper = await getComponent({color});
      expect(wrapper.find(LinearProgress)).toExist();
      expect(wrapper.find(LinearProgress).prop("color")).toEqual(color);
    });

    it("does not renders if loading prop is false", async () => {
      const loading = false;
      const wrapper = await getComponent({loading});
      expect(wrapper.find(LinearProgress)).not.toExist();
    });
  });

  const getComponent = props => {
    const parsedProps = extend(getMockProps(), props);

    return shallow(<Preloader {...parsedProps} />);
  };

  const renderComponent = props => {
    const parsedProps = extend(getMockProps(), props);

    return renderer.create(<Preloader {...parsedProps} />).toJSON();
  };

  const getMockProps = () => ({
    classes: {
      root: "Preloader-root-122",
    },
    loading: true,
  });
});
