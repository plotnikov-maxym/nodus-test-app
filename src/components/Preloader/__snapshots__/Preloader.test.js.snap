// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Preloader /> Preloader component matches snapshot 1`] = `
<div
  className="root-122"
>
  <div
    className="MuiLinearProgress-root-1 MuiLinearProgress-colorPrimary-2 MuiLinearProgress-indeterminate-5"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar-11 MuiLinearProgress-barColorPrimary-12 MuiLinearProgress-bar1Indeterminate-14"
      style={Object {}}
    />
    <div
      className="MuiLinearProgress-bar-11 MuiLinearProgress-barColorPrimary-12 MuiLinearProgress-bar2Indeterminate-17"
      style={Object {}}
    />
  </div>
</div>
`;
